state/core/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityViewUpdater.java

```diff
@@ -51,45 +55,62 @@
     }
   }
 
+  <K extends Entity.Key<S>, S extends Specification> Optional<Entity<K, S>> purge(K key) {
+    val stateEntity = Optional.ofNullable(entities.get(key))
+      .filter(it -> it.deleted);
+    stateEntity.ifPresent(it -> {
+      entities.remove(key);
+      log.debug("Purged entity for key={}", key);
+    });
+    return stateEntity.map(it -> (Entity<K, S>) it.entity);
+  }
+
   private <K extends Entity.Key<S>, S extends Specification> Entity<K, S> update(SpecificationEvent<K, S> event) {
-    val oldEntity = (Entity<K, S>) entities.get(event.getKey());
-    val status = Optional
-        .ofNullable(oldEntity)
-        .map(Entity::getStatus)
-        .orElseGet(DefaultStatus::new);
+    val oldEntity = (Entity<K, S>) getExistingEntity(event.getKey());
+    val status = Optional.ofNullable(oldEntity)
+      .map(Entity::getStatus)
+      .orElseGet(DefaultStatus::new);
     val entity = new Entity<>(event.getKey(), event.getSpecification(), status);
-    entities.put(event.getKey(), entity);
+    entities.put(event.getKey(), existing(entity));
     log.debug("Updated {} with {}", event.getKey(), event.getSpecification());
     return oldEntity;
   }
 
   private <K extends Entity.Key<S>, S extends Specification> Entity<K, S> update(StatusEvent<K, S> event) {
-    val oldEntity = (Entity<K, S>) entities.get(event.getKey());
+    val oldEntity = (Entity<K, S>) getExistingEntity(event.getKey());
     if (oldEntity == null) {
       log.warn("Received status {} non existent entity {}", event.getStatusEntry().getName(), event.getKey());
       return null;
     }
     val entity = new Entity<>(event.getKey(), oldEntity.getSpecification(), oldEntity.getStatus().with(event.getStatusEntry()));
-    entities.put(event.getKey(), entity);
+    entities.put(event.getKey(), existing(entity));
     log.debug("Updated {} with {}", event.getKey(), event.getStatusEntry());
     return oldEntity;
   }
 
   private <K extends Entity.Key<S>, S extends Specification> Entity<K, S> delete(SpecificationDeletionEvent<K, S> event) {
-    val oldEntity = (Entity<K, S>) entities.remove(event.getKey());
+    val oldEntity = (Entity<K, S>) getExistingEntity(event.getKey());
+    entities.put(event.getKey(), deleted(oldEntity));
     log.debug("Deleted entity for {}", event.getKey());
     return oldEntity;
   }
 
   private <K extends Entity.Key<S>, S extends Specification> Entity<K, S> delete(StatusDeletionEvent<K, S> event) {
-    val oldEntity = (Entity<K, S>) entities.get(event.getKey());
+    val oldEntity = (Entity<K, S>) getExistingEntity(event.getKey());
     if (oldEntity == null) {
       log.warn("Received status deletion {} for non existent entity {}", event.getStatusName(), event.getKey());
       return null;
     }
     val entity = oldEntity.withStatus(oldEntity.getStatus().without(event.getStatusName()));
-    entities.put(event.getKey(), entity);
+    entities.put(event.getKey(), existing(entity));
```

Comment:
Should it be this?
```suggestion
    entities.put(event.getKey(), deleted(entity));
```

Reply:
Eagle eyed 👍 

So hopefully there are test cases to cover this, but no I don't believe this should be. 

In this method we are deleting the Status rather than deleting the Entity. So we want to remove the Status from an existing Entity, but not actually delete that Entity.

Another minor point, it's not possible to delete a status on a delete entity, which I believe is the correct functionality.